src/test/bufferlist.cc

```diff
@@ -2369,6 +2369,25 @@ TEST(BufferList, splice) {
     EXPECT_EQ((unsigned)11, bl.length());
     EXPECT_EQ(0, ::memcmp("fghi", other.c_str(), other.length()));
   }
+
+  {
+    /* bug replicator, to test rebuild_aligned_size_and_memory() in this
+     * special scenarios */
+    bufferlist bl1;
+    bl.clear();
+    std::vector<int> data(30);
+    bufferptr ptr(buffer::create_aligned(16384, 4096));
+
+    encode(data, bl);
+    bl.append_zero(4096 - bl.length());
+    bl.append(ptr);
+    bl.splice(0, 8192, &bl1);
+    EXPECT_EQ(bl.get_num_buffers(), 2);
+    EXPECT_EQ(bl.back().length(), 0);
+    bl.rebuild_aligned_size_and_memory(4096, 4096, 1024);
```

Comment:
Is `max_buffers=1024` relevant for reproducing?  Wouldn't `bl.rebuild_aligned(4096)` be sufficient?